azure_data_cosmos_engine/src/query/producer/mod.rs

```diff
@@ -51,6 +59,49 @@ pub fn create_partition_state(
     partitions
 }
 
+pub fn create_query_chunk_states(
+    query_chunks: &Vec<HashMap<String, Vec<(usize, String, String)>>>,
+) -> Vec<QueryChunkState> {
+    let mut chunk_states = Vec::with_capacity(query_chunks.len());
+
+    for i in 0..query_chunks.len() {
+        let map = &query_chunks[i];
+
+        let key: String = map
+            .keys()
+            .next()
+            .cloned()
+            .expect("Expected exactly one key in each hashmap");
+
+        let query = create_query_chunk_query(&key, &map);
+        tracing::trace!(?query, "created query chunk query from partitioned items");
+
+        chunk_states.push(QueryChunkState::new(i, key, query));
+    }
+    chunk_states
+}
+
+fn create_query_chunk_query(
+    key: &str,
+    map: &HashMap<String, Vec<(usize, String, String)>>,
```

Comment:
Wondering why take the `key` and the `map` here instead of just taking a borrow of the `identities` `Vec` directly?